pkg/generate/app/cmd/newapp.go

```diff
@@ -222,26 +224,56 @@ func (c *AppConfig) SetOpenShiftClient(osclient client.Interface, OriginNamespac
 // AddArguments converts command line arguments into the appropriate bucket based on what they look like
 func (c *AppConfig) AddArguments(args []string) []string {
 	unknown := []string{}
+	c.ArgumentClassificationErrors = map[string]error{}
 	for _, s := range args {
-		switch {
-		case cmdutil.IsEnvironmentArgument(s):
+		if len(s) == 0 {
+			continue
+		}
+
+		// had to move off of elegant switch to capture errors for future display, but mimic
```

Comment:
It certainly is .... between your comment and the exchange I had with @bparees, I believe the explanation has been sufficiently proliferated :-)

deleted with next push